### PR TITLE
fix: improving feature util test coverage

### PR DESCRIPTION
**Pull request by drduhe**
**Issue #, if available:** n/a

### Notes
This PR improves the robustness of the `features_to_image_shapes` function by allowing it to handle mixed valid and invalid features without terminating. Additionally, it expands the test coverage to ensure more thorough testing of edge cases and potential failure points in the feature conversion process. 

### **Key Changes:**
1. **Enhancement of `features_to_image_shapes`:**
   - Updated the function to skip invalid features instead of raising a `ValueError`. This allows the processing of mixed datasets, where valid features can still be converted even if some are malformed or unsupported.
   - Added logging to inform users when a feature is skipped due to errors, improving traceability and debugging.

2. **New and Updated Unit Tests:**
   - **Refactored `test_features_conversion_no_geometry` and `test_features_conversion_unsupported_type`:**
     - Adjusted these tests to align with the new behavior of `features_to_image_shapes`. They now verify that invalid features are skipped, rather than expecting exceptions.
   - **Improved `test_features_conversion_mixed`:**
     - Verified that only valid features are processed, ensuring that the function can handle datasets with a mix of valid and invalid inputs.
     - Enhanced with additional logging to aid in debugging.
   - **New Test for Basic Valid Feature Conversion:**
     - Added `test_features_conversion_basic_point` to confirm that simple, valid features are correctly processed and converted.
   - **New Test for `convert_nested_coordinate_lists`:**
     - Ensured the function behaves correctly when handling single vs. nested coordinates.
     - Validated that the conversion function handles input as expected without altering valid data.
   - **Additional Debugging Tests:**
     - Added more logging within the `features_to_image_shapes` function to trace feature processing, helping to identify why valid features might be skipped.

3. **Improved Error Handling:**
   - The function now uses `try-except` blocks to catch and log errors during feature processing. Invalid features are logged and skipped, instead of halting the entire process. 

### Checklist

Before you submit a pull request, please make sure you have the following:
- [x] Code changes are compact and well-structured to facilitate easy review
- [x] Changes are documented in the README.md and other relevant documentation pages
- [x] PR title and description accurately reflect the changes and are detailed enough for historical tracking
- [x] PR contains tests that cover all new code and the code has been manual tested
- [x] All new dependencies are declared (if any), and no unnecessary libraries are added
- [x] Performance impacts (if any) of the changes are evaluated and documented
- [x] Security implications of the changes (if any) are reviewed and addressed
- [x] I have read the [Contributing Guidelines](../CONTRIBUTING.md) and agree to follow the [Code of Conduct](../CODE_OF_CONDUCT.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/aws/osml/model_runner/inference/feature_utils.py [hunk 1]
```diff
@@ -34,35 +31,41 @@ def features_to_image_shapes(sensor_model: SensorModel, features: List[Feature])
...
-        elif isinstance(feature_geometry, MultiLineString):
-            shapes.append(shapely.geometry.MultiLineString(image_coords))
-        elif isinstance(feature_geometry, MultiPolygon):
-            shapes.append(shapely.geometry.shape(feature_geometry))
-        else:
-            # Unlikely to get here as we're handling all valid geojson types but if the spec
-            # ever changes or if a consumer passes in a custom dictionary that isn't valid
-            # we want to handle it gracefully
-            raise ValueError("Unable to convert feature due to unrecognized or invalid geometry")
+        try:
```
Comment (edparris) [inline src/aws/osml/model_runner/inference/feature_utils.py:40]: Why was this function changed to skip geometries instead of raise exceptions?

Comment (drduhe) [reply src/aws/osml/model_runner/inference/feature_utils.py:40]: If there are valid features in the list do we not want to process them instead of dropping the entire thing and warn the user that a subset was missed?

Comment (drduhe) [reply src/aws/osml/model_runner/inference/feature_utils.py:51]: Updated to include logic to toggle the skipping of bad features.
